packages/common/src/api/tan-query/jupiter/useSwapTokens.ts

```diff
@@ -0,0 +1,375 @@
+import { createCloseAccountInstruction } from '@solana/spl-token'
+import {
+  PublicKey,
+  TransactionInstruction,
+  VersionedTransaction
+} from '@solana/web3.js'
+import { useMutation, useQueryClient } from '@tanstack/react-query'
+
+import { useGetCurrentUser } from '~/api'
+import { useAudiusQueryContext } from '~/audius-query'
+import { Feature } from '~/models'
+import {
+  convertJupiterInstructions,
+  getJupiterQuoteByMint,
+  jupiterInstance
+} from '~/services/Jupiter'
+
+import { QUERY_KEYS } from '../queryKeys'
+
+import { USER_BANK_MANAGED_TOKENS } from './constants'
+import {
+  SwapErrorType,
+  SwapStatus,
+  SwapTokensParams,
+  SwapTokensResult
+} from './types'
+import {
+  addAtaToUserBankInstructions,
+  addUserBankToAtaInstructions,
+  findActualJupiterDestination,
+  findJupiterTemporarySetupAta,
+  updateJupiterAtaCreationFeePayer
+} from './utils'
+
+/**
+ * Hook for executing token swaps using Jupiter.
+ * Swaps any supported SPL token (or SOL) for another.
+ */
+export const useSwapTokens = () => {
+  const queryClient = useQueryClient()
+  const { solanaWalletService, reportToSentry, audiusSdk } =
+    useAudiusQueryContext()
+  const { data: user } = useGetCurrentUser({})
+
+  return useMutation<SwapTokensResult, Error, SwapTokensParams>({
+    mutationFn: async (params): Promise<SwapTokensResult> => {
+      const {
+        inputMint: inputMintUiAddress,
+        outputMint: outputMintUiAddress,
+        amountUi
+      } = params
+      const slippageBps = params.slippageBps ?? 50
+      const wrapUnwrapSol = params.wrapUnwrapSol ?? true
+
+      let quoteResult
+      let signature: string | undefined
+      const instructions: TransactionInstruction[] = []
+
+      try {
+        // ---------- 1. Initialize Dependencies & Wallet ----------
+        const [sdk, keypair] = await Promise.all([
+          audiusSdk(),
+          solanaWalletService.getKeypair()
+        ])
+
+        if (!keypair) {
+          return {
+            status: SwapStatus.ERROR,
+            error: {
+              type: SwapErrorType.WALLET_ERROR,
+              message: 'Wallet not initialised'
+            }
+          }
+        }
+        const userPublicKey = keypair.publicKey
+        const feePayer = await sdk.services.solanaRelay.getFeePayer()
```

Comment:
better to do `sdk.services.solanaClient.getFeePayer()` - shouldn't matter whether it's relayed or not